tests/lib/c_lib/src/main.c

```diff
@@ -468,24 +499,30 @@ void test_checktype(void)
  */
 void test_memstr(void)
 {
+	int i, j, ret;
 	static const char str[] = "testfunction";
-	int ch1 = 'e', ch2 = 'z';
-	char arr[10], move_arr[6] = "12123";
-	static const char num[6] = "12121";
-
-	zassert_equal(memchr(str, ch1, strlen(str)), str + 1,
-		     "memchr 'testfunction' 'e' ");
-	zassert_equal(memchr(str, ch2, strlen(str)), NULL,
-		     "memchr 'testfunction' 'z'");
-
-	zassert_equal(memcpy(arr, num, sizeof(num)), arr, "memcpy error");
-	zassert_equal(memcmp(num, arr, sizeof(num)), 0,
-		     "memcpy failed");
+	char   arr[1000], move_arr[6] = "12123";
+	static const char num[30] = "1234567891234567891234";
```

Comment:
Here especially `[]` might be preferred.  It depends on why you're allocating 30 bytes to store a string that's 25 characters long (including terminating NUL)?

Reply:
ok, I agree with you , I will update it.